flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  description = ''
+    Flake for the cf-cli-plugin for app-autoscaler
+
+    For more on app-autoscaler, see: <https://github.com/cloudfoundry/app-autoscaler-release>
+  '';
+
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-23.11;
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = nixpkgs.lib.genAttrs supportedSystems;
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+    in flake-utils.lib.eachSystem supportedSystems (system: {
+      packages = {
+        app-autoscaler-cli-plugin = nixpkgsFor.${system}.buildGoModule rec {
+          pname = "app-autoscaler-cli-plugin";
+          version =
+            let
+              lastModifiedDate = self.lastModifiedDate or self.lastModified or "19700101";
+            in "${builtins.substring 0 8 lastModifiedDate}-dev";
```

Comment:
Do we get a new `version` if anything except the `flake.nix` / `flake.lock` changes?

Reply:
Good point. No, we don't: <https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake#flake-reference-attributes>

What could be better? 🤔 
 + Use current time: That's not reproducable. Identical code produced by identical builders will have two different versions.
 + Just use "dev"? That will work!
 + Stay as we are? At least compared to the option above, we will get a change in the version when the dependencies change (due to implication on vendorHash).

Reply:
It feels like none of the options are ideal as long as we don't have automation which takes care of the versioning. 

> Just use "dev"? That will work!
I vote for 👆
* most simple solution
* everbody can get the version information just by looking at the `flake.nix` file
* not much going on in this repo anyhow so consuming always the latest `dev`-version is not hurting